wallet_transaction.go

```diff
@@ -68,7 +68,10 @@ func (bmr *WalletTransactionRequest) Create(ctx context.Context, walletTransacti
 		return nil, err
 	}
 
-	walletTransactionResult := result.(*WalletTransactionResult)
+	walletTransactionResult, ok := result.(*WalletTransactionResult)
+	if !ok {
+		return nil, err
```

Comment:
the err will be from bmr.client.Post and so will be nil ;)
IMHO, since you do a lot of type assertion, an easy fix would be to define
var ErrTypeAssert = errors.New("failed to cast result") at a global level and return this after every !ok